source/reference/minio-mc/mc-support-diagnostics.rst

```diff
@@ -0,0 +1,132 @@
+
+The resulting health report is intended for use by MinIO Engineering via |subnet| and may contain internal or private data points, such as hostnames.
+Exercise caution before sending a health report to a third party or posting the health report in a public forum.
+
+MinIO recommends that you run the health diagnostics when first provisioning the cluster and again at any failure scenario.
+
+This diagnostic test runs as a one-shot test and can run for a long time.
+This test uses the majority of the available resources on the system. 
+Use care when running this diagnostic test to debug failure scenario to prevent larger outages. 
+
+Use the :mc-cmd:`mc support diagnostics` command to trigger the diagnostic test.
+You can then upload the results of the test can to the MinIO subscription network, |subnet|.
+
+
+Sample Output
+~~~~~~~~~~~~~
+
+.. code-block:: shell
+
+   ● Admin Info ... ✔ 
+   ● CPU ... ✔ 
+   ● Disk Hardware ... ✔ 
+   ● Os Info ... ✔ 
+   ● Mem Info ... ✔ 
+   ● Process Info ... ✔ 
+   ● Config ... ✔ 
+   ● Drive ... ✔ 
+   ● Net ... ✔ 
+   *********************************************************************************
+                                   WARNING!!
+        ** THIS FILE MAY CONTAIN SENSITIVE INFORMATION ABOUT YOUR ENVIRONMENT ** 
+        ** PLEASE INSPECT CONTENTS BEFORE SHARING IT ON ANY PUBLIC FORUM **
+   *********************************************************************************
+   mc: Health data saved to dc-11-health_20220511053323.json.gz
+
+The gzipped output contains the requested health information.
+
+
+Examples
+--------
+
+Upload Health Data for a Cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generate health data for a MinIO cluster at alias ``minio1`` for transmission to |SUBNET|.
+
+.. code-block:: shell
+   :class: copyable
+
+   mc support diag minio1
+
+Generate Health Data for a Cluster to Upload Manually
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generate a diagnostic report for a MinIO deployment at alias ``minio2`` and save it for manual upload to |SUBNET|:
+
+.. code-block:: shell
+   :class: copyable
+
+   mc support diag minio2 --airgap
+
+
+
+Syntax
+------
+      
+The command has the following syntax:
+
+.. code-block:: shell
+
+   mc [GLOBALFLAGS] support diag       \
+                            ALIAS      \
+                            [--full]   \
+                            [--name]   \
+                            [--airgap] 
+
+
+Parameters
+~~~~~~~~~~
+
+.. mc-cmd:: ALIAS
+   :required:
+
+   The :ref:`alias <alias>` of the MinIO deployment.
+
+.. mc-cmd:: --full
+   :optional:
+   
+   Include long running diagnostics.
+
+   This option takes longer to generate.
+
+.. mc-cmd:: --name
+   :optional:
+
+   Specify a name other than the deployment's alias to associate to the MinIO cluster in |SUBNET|.
+
+   Use ``--name <value>``, replacing ``<value>`` with the name to use.
```

Comment:
The `--name` flag was meaningful with `support diag` command when it was possible to run and upload diagnostics even before registering the cluster with SUBNET (it used to internally trigger registration before the upload).

Now `support diag` errors out if the cluster is not registered. This means that that cluster must be already registered when `suport diag` is being run, which means it already has a name on SUBNET, and in such cases the `--name` flag is meaningless as it will be ignored by SUBNET.

I will send a PR in `mc` to remove this flag. I think we need not add documentation for this flag.